week-1-ts-ecosystem/w01e03-object-oriented-structure/src/Cart.ts

```diff
@@ -0,0 +1,53 @@
+import { AuctionProduct, BuyNowProduct, FreeProduct } from "./Products";
+
+type ProductTypes = BuyNowProduct | FreeProduct | AuctionProduct;
+
+export class Cart<Product extends ProductTypes> {
+  #products = new Map<string, Product>();
+
+  getById(id: string) {
+    return this.#products.get(id);
+  }
+
+  addProduct(product: Product) {
+    if (this.#products.has(product.id)) {
```

Comment:
Dodanie tego samego produktu - powinno sumować `.amount` i jest możliwe.
Wyobraź sobie że ktoś klika "Dodaj do koszyka" 2x ten sam produkt. To podobna sytuacja.